### PR TITLE
[2.0] Remove list_run_infos

### PR DESCRIPTION
**Pull request by dbczumar**
Signed-off-by: dbczumar <corey.zumar@databricks.com>

## What changes are proposed in this pull request?

Remove list_run_infos

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

## Release Notes

Remove list_run_infos API

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [X] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [X] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: tests/projects/test_projects.py [hunk 1]
```diff
@@ -133,8 +133,8 @@ def test_run_local_git_repo(local_git_repo, local_git_repo_uri, use_start_run, v
     # Validate run contents in the FileStore
     run_id = submitted_run.run_id
     mlflow_service = MlflowClient()
-    run_infos = mlflow_service.list_run_infos(
-        experiment_id=FileStore.DEFAULT_EXPERIMENT_ID, run_view_type=ViewType.ACTIVE_ONLY
+    run_infos = mlflow_service.search_runs(
+        [FileStore.DEFAULT_EXPERIMENT_ID], run_view_type=ViewType.ACTIVE_ONLY
     )
     assert len(run_infos) == 1
     store_run_id = run_infos[0].run_id
```
Comment (BenWilson2) [inline tests/projects/test_projects.py:140]: Should these accessors be `run_infos[0].info.run_id`?

**Comment by BenWilson2** [comment]
Are there going to be any issues with having the return signature change from `RunInfo` to `Run`? (will we need an explicit docs entry to explain this to users; e.g. "MLflow2.0 has removed the deprecated list_run_infos() API in favor of using search_runs(). Note that the return signature will now return `List[Run]` as opposed to `List[RunInfo]`" ?)
Looks like this is the cause of all of the test failures...

**Comment by BenWilson2** [comment]
Should SqlAlchemy store allow for a null / missing filter string in the `search_runs()` API through the use of a default "" for the filtering query as the abstract definition has (and other stores utilize)?

File: tests/projects/test_projects.py (hunk 1 above)
Comment (dbczumar) [reply tests/projects/test_projects.py]: Yep! thanks Ben!

**Comment by dbczumar** [comment]
> Are there going to be any issues with having the return signature change from `RunInfo` to `Run`? (will we need an explicit docs entry to explain this to users; e.g. "MLflow2.0 has removed the deprecated list_run_infos() API in favor of using search_runs(). Note that the return signature will now return `List[Run]` as opposed to `List[RunInfo]`" ?) Looks like this is the cause of all of the test failures...

I think it's sufficient to publish breaking change info. search_runs is fairly well documented - hopefully users should be able to migrate

> Should SqlAlchemy store allow for a null / missing filter string in the `search_runs()` API through the use of a default "" for the filtering query as the abstract definition has (and other stores utilize)?

Abstract store doesn't define a default filter string https://github.com/mlflow/mlflow/blob/d4a2475df1c3c90ff0d6d24f2e18540e52172019/mlflow/store/tracking/abstract_store.py#L248, so I think it's okay for now.

**Review by BenWilson2 — APPROVED** [review]
LGTM!

**Comment by dbczumar** [comment]
Remaining failures are caused by `sqlparse` upgrade to 0.4.3 and are also present on master. Merging :)

**Pull request merged**